### PR TITLE
Fix controller networkdata secret generation

### PR DESCRIPTION
**Pull request by stuggi**
The networkdata rendering is running for all controllers with the
same ControllerIP template parameter which results in with multiple
controllers that all have the same ip on the osp network. Now we
make sure that the secret gets rendered with uniq ips.

### CODE REVIEW

File: controllers/controllervm_controller.go [hunk 1]
```diff
@@ -223,6 +227,7 @@ func (r *ControllerVMReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	if instance.Status.BaseImageDVReady {
 		// Generate the Contoller networkdata secrets
 		for _, ctl := range controllerDetails {
+			r.Log.Info(fmt.Sprintf("DEBUG: %v", ctl))
```
Comment (dprince) [inline controllers/controllervm_controller.go:230]: do we want to leave this in?

File: controllers/controllervm_controller.go [hunk 2]
```diff
@@ -152,34 +157,33 @@ func (r *ControllerVMReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
...
+			DomainName:        hostname,
+			DomainNameUniq:    fmt.Sprintf("%s-%s", hostname, instance.UID[0:4]),
+			IPAdress:          fmt.Sprintf("192.168.25.1%d", i),
+			NetworkDataSecret: networkDataSecretName,
+		}
 
-	err = common.EnsureSecrets(r, instance, sts, &envVars)
-	if err != nil {
-		return ctrl.Result{}, nil
+		templateParameters["ControllerIP"] = controllerDetails[hostname].IPAdress
```
Comment (dprince) [inline controllers/controllervm_controller.go:170]: I had noticed this last week. No need to fix it here but I think we should rename this IPAddress.

**Review by dprince — APPROVED** [review]

**Comment by abays** [comment]
This looks good to me

File: controllers/controllervm_controller.go (hunk 1 above)
Comment (stuggi) [reply controllers/controllervm_controller.go:230]: no, removing it

File: controllers/controllervm_controller.go (hunk 2 above)
Comment (stuggi) [reply controllers/controllervm_controller.go:170]: yes sounds good. we can do it with the IPAM integration

**Pull request merged**